### PR TITLE
Add a config to control if authgear verify the signature of saml elements from SP

### PR DESCRIPTION
**Pull request by tung2744**
ref DEV-2145

### CODE REVIEW

**Comment by louischan-oursky** [comment]
Test failed.

File: pkg/lib/config/saml.go [hunk 1]
```diff
@@ -39,7 +39,8 @@ var _ = Schema.Add("SAMLServiceProviderConfig", `
 		"assertion_valid_duration":  { "$ref": "#/$defs/DurationString" },
 		"slo_enabled": { "type": "boolean" },
 		"slo_callback_url": { "type": "string", "format": "uri" },
-		"slo_binding": { "$ref": "#/$defs/SAMLSLOBinding" }
+		"slo_binding": { "$ref": "#/$defs/SAMLSLOBinding" },
+		"should_verify_signature": { "type": "boolean" }
```
Comment (louischan-oursky) [inline pkg/lib/config/saml.go:43]: Rename this to `signature_verification_enabled`

File: pkg/lib/saml/service.go [hunk 2]
```diff
@@ -563,10 +563,13 @@ func (s *Service) IssueLogoutResponse(
...
 	samlElementXML string) error {
+	if !sp.ShouldVerifySignature {
+		return nil
+	}
 	certs, ok := s.SAMLSpSigningMaterials.Resolve(sp)
 	if !ok || len(certs.Certificates) == 0 {
-		// Signing cert not configured, nothing to verify
-		return nil
+		// This should be prevented by config validation. Therefore it is a programming error.
+		panic("SP certificates not configured but signature verification is required")
```
Comment (louischan-oursky) [inline pkg/lib/saml/service.go:572]: [panic with error](https://github.com/authgear/authgear-server/issues/3626#:~:text=%C2%A0%234804%20(comment)-,panic%20with%20error,-Instead%20of%20panic). I will study whether there is golangci-lint rule to address this.

File: pkg/lib/saml/service.go [hunk 3]
```diff
@@ -601,10 +604,13 @@ func (s *Service) VerifyExternalSignature(
...
 	signature string) error {
+	if !sp.ShouldVerifySignature {
+		return nil
+	}
 	certs, ok := s.SAMLSpSigningMaterials.Resolve(sp)
 	if !ok || len(certs.Certificates) == 0 {
-		// Signing cert not configured, nothing to verify
-		return nil
+		// This should be prevented by config validation. Therefore it is a programming error.
+		panic("SP certificates not configured but signature verification is required")
```
Comment (louischan-oursky) [inline pkg/lib/saml/service.go:613]: [panic with error](https://github.com/authgear/authgear-server/issues/3626#:~:text=%C2%A0%234804%20(comment)-,panic%20with%20error,-Instead%20of%20panic)

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

File: pkg/lib/saml/service.go (hunk 2 above)
Comment (tung2744) [reply pkg/lib/saml/service.go:572]: Just curious, what is the advantage of panic with error instead of string?

**Comment by tung2744** [comment]
Updated, thanks!

File: pkg/lib/saml/service.go (hunk 2 above)
Comment (louischan-oursky) [reply pkg/lib/saml/service.go]: Updated that link to include this rationale.

**Comment by tung2744** [comment]
Fixed, thanks!

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**